### PR TITLE
Fix the PUNCTUATION method issue with non ascii characters

### PR DESCRIPTION
**Pull request by purbon**
Fixed, and make more standard, the PUNCTUATION issue (#4) with non ascii characters being not filtered.

_Note_

The situation where the characters that one might think are punctuation are actually symbols, for example the ```<``` and ```>```. In the case of the ```[[:punct:]]``` POSIX syntax this characters are not deal with, so it's better to use the more general UNICODE syntax for that.

### CODE REVIEW

**Comment by purbon** [comment]
@jsvd  @colinsurprenant @suyograo this need a review, the change is staying here for too long :-)

File: lib/logstash/filters/fingerprint.rb [hunk 1]
```diff
@@ -62,7 +62,10 @@ def filter(event)
       when "PUNCTUATION"
         @source.sort.each do |field|
           next unless event.include?(field)
-          event[@target] = event[field].tr('A-Za-z0-9 \t','')
+          # In order to keep some backwards compatibility we should use the unicode version
+          # of the regexp because the POSIX one ([[:punct:]]) left some unwanted characters unfiltered (Symbols).
+          # gsub(/[^[:punct:]]/,'') should be equivalent to gsub(/[^[\p{P}\p{S}]]/,''), but not 100% in JRuby.
+          event[@target] = event[field].gsub(/[^[\p{P}\p{S}]]/,'')
```
Comment (jsvd) [inline lib/logstash/filters/fingerprint.rb]: an alternative is: `/[[:alnum:]]|[[:blank:]]/`

**Comment by jsvd** [comment]
LGTM, minor suggestion. rebase and :shipit:

File: lib/logstash/filters/fingerprint.rb (hunk 1 above)
Comment (purbon) [inline lib/logstash/filters/fingerprint.rb]: I did no recent test, but as far as I remember this is also does not work, as it translated to the other non posix version. Some edge case of this kind of expressions in ruby unicode. I'm curious, did you try it out like this?
